### ISSUE TITLE
implement tie/breakers

### ISSUE THREAD

**Issue by cmavelis** [cmavelis/zug#109]
when both players reach 4 pts on the same turn, there should be a tie or a tiebreaker. As of now, first player wins because they get checked first

**Comment by cmavelis** [comment]
default tie added in #113 
tiebreakers later

**Comment by bpdeforest** [comment]
If one of the tiebreakers is going to be "whose piece scored first," I think it would need to be clarified whether or not it matters if a piece is pushed after having initially arrived at a scoring position.  It's possible for a piece to enter a scoring position, move or get pushed out of the scoring position, and then move or get pushed back into a scoring position.  It's also possible for a piece just to get pushed sideways on the final row such that it never leaves a scoring position, though it leaves the square it was originally scoring on

**Comment by bpdeforest** [comment]
I think there are three versions of how to determine when a piece arrived for the purposes of deciding "whose piece scored first":
1) a piece arrives at any scoring position on the board and ends the turn at any scoring position on the board, and the time that matters is that of the piece's first arrival at a scoring position
2) a piece arrives at a particular scoring position and ends the turn at that same scoring position after having moved/been pushed, and the time that matters is that of the piece's first arrival at that particular scoring position
3) a piece arrives at any scoring position on the board and ends the turn at any scoring position on the board, and the time that matters is that of the piece's final arrival at a scoring position

**Comment by cmavelis** [comment]
my thought would be option 1 makes the most sense, you?

**Comment by bpdeforest** [comment]
I can see arguments for and against all of them, and I'm not really sure which one I favor.  One of the arguments against #1 would be that, when you're reviewing the moves that took place during the final round, the winning piece according to #1 could have 1) moved/been moved into a scoring position, 2) moved/been moved out of scoring position, and then 3) moved/been moved back into a scoring position.  If I'm the opponent, and I can see on the replay that I scored my piece after 2) has taken place but before 3) has, I think I'd be annoyed.  I suppose that particular issue could be addressed by tweaking #1 slightly: "a piece arrives at any scoring position on the board and remains in a scoring position through the end of the turn, and the time that matters is that of the piece's first entering into a sustained state of being in a scoring position."  (There's probably a simpler way to write that.)

I think the other argument against #1 is that it might potentially be giving an advantage to the player who has the lower-numbered pieces on the board.  I could imagine someone thinking that the two players on a 3-3 board should have an equal chance of making interesting and potentially winning plays during the final turn, and making the tiebreaker about when a piece _first_ gets to its position could be providing one player with an inherent advantage—an advantage that they got at least in part by chance.  I'm not actually sure if this is true, though: I assume you'd need a supercomputer to test all the permutations to know which configurations of piece numbers are more likely to result in potential wins on a 3-3 board, and it might turn out that having the lower-numbered pieces isn't advantageous.  And it might turn out that #2 and #3 are as bad as or worse in regard to giving each player an equal chance.  And, of course, some might say that the final 3-3 board is the result of everything that came before it, so there's no reason to think that players should be on an equal footing for it.

In short, I'm not really sure what makes the most sense.  I think the revised version of #1 is probably better than the original version I'd suggested for #1, but I'm not sure that everyone would agree.
